legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/runtime/connection/authenticationStrategy.pure

```diff
@@ -70,4 +70,20 @@ Class meta::pure::alloy::connections::alloy::authentication::GCPWorkloadIdentity
 {
     serviceAccountEmail : String[1];
     additionalGcpScopes: String[*];
-}
\ No newline at end of file
+}
+
+Class
+{doc.doc = 'Private key based authentication on AWS'}
+meta::pure::alloy::connections::alloy::authentication::AwsPKAuthenticationStrategy extends meta::pure::alloy::connections::alloy::authentication::AuthenticationStrategy
+{
+    secretArn: String[1];
+    user: String[1];
+}
+
+Class
+{doc.doc = 'OAuth-based authentication on AWS'}
+meta::pure::alloy::connections::alloy::authentication::AwsOAuthAuthenticationStrategy extends meta::pure::alloy::connections::alloy::authentication::AuthenticationStrategy
```

Comment:
It will be nice if we can unify how we represent secrets. 


So far we have metamodels that simply model secrets as freeform strings (e.g privateKeyVaultReference ) and the Java execution code interprets the strings. E.g PropertiesVault vs EnvironmentVault etc

Since we do not have any type information, to lookup a secret, we have ask every registered vault implementation.
If we had some type information, we can just ask the vault implementation for that type.

So something like 
```
{
    secret : aws/xxxx
    secret : gcp/xxxx
    secret : properties/xxxx
    secret : {
          type : aws
          reference: xxxxx 
    }
}
```